pykafka/protocol.py

```diff
@@ -125,6 +127,45 @@ def raise_error(self, err_code, response):
                 clsname, response))
 
 
+class MessageOffset(object):
+    """A message offset described as a pair:
+    the offset of the message or its containing compressed message,
+    and the offset of the submessage in its containing message, if any."""
+    __slots__ = ("main", "sub")
+
+    def __init__(self, main_offset, sub_offset=None):
+        self.main = main_offset
+        self.sub = sub_offset
+
+    def can_precede(self, other, compressed_topic):
```

Comment:
I think you might mean `is_compacted_topic` here, since a topic can contain both compressed and uncompressed messages.